### PR TITLE
Add swarm section to 1.12.1 changelog

### PR DESCRIPTION
**Pull request by aaronlehmann**
This adds the main swarm fixes for 1.12.1 to the changelog. It also changes the release date in the changelog.

ping @tiborvass @docker/core-swarmkit-maintainers

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -84,6 +84,18 @@ systemctl restart docker` to reload changes and (re)start the docker daemon.
 - Fix an issue where a volume mount could inadvertently create a bind mount [#25309](https://github.com/docker/docker/pull/25309)
 - `docker service create --mount type=bind,...` now correctly validates if the source path exists, instead of creating it [#25494](https://github.com/docker/docker/pull/25494)
 
+### Swarm
+
+- Fix an issue where the swarm can get stuck electing a new leader after quorum is lost [#25055](https://github.com/docker/docker/issues/25055)
+- Fix unwanted rescheduling of containers after a leader failover [#25017](https://github.com/docker/docker/issues/25017)
+- Change swarm root CA key to P256 curve [#25767](https://github.com/docker/docker/pull/25767)
```
Comment (dongluochen) [inline CHANGELOG.md]: [swarmkit#1376](https://github.com/docker/swarmkit/pull/1376)

**Comment by tonistiigi** [comment]
LGTM

**Comment by dongluochen** [comment]
SGTM (not a maintainer).

File: CHANGELOG.md [hunk 2]
```diff
@@ -84,6 +84,18 @@ systemctl restart docker` to reload changes and (re)start the docker daemon.
 - Fix an issue where a volume mount could inadvertently create a bind mount [#25309](https://github.com/docker/docker/pull/25309)
 - `docker service create --mount type=bind,...` now correctly validates if the source path exists, instead of creating it [#25494](https://github.com/docker/docker/pull/25494)
 
+### Swarm
```
Comment (tiborvass) [inline CHANGELOG.md]: @aaronlehmann can you please move this above Volume section? trying to make it alphabetical, although it's quite inconsistent through releases.

Comment (aaronlehmann) [inline CHANGELOG.md]: done

**Comment by justincormack** [comment]
LGTM

**Comment by abronan** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**